weston-ivi-shell/src/ivi-controller.c

```diff
@@ -1623,10 +1756,35 @@ surface_event_configure(struct wl_listener *listener, void *data)
         lyt->commit_changes();
     }
 
-    wl_list_for_each(not, &ivisurf->notification_list, layout_link) {
-        ctrl = wl_resource_get_user_data(not->resource);
-        send_surface_event(ctrl, ivisurf->layout_surface, surface_id, ivisurf->prop,
-                           IVI_NOTIFICATION_CONFIGURE);
+    if (shell->bkgnd_surface_id &&
+       (shell->bkgnd_surface_id == surface_id)) {
+        float red, green, blue, alpha;
+
+        w_surface = lyt->surface_get_weston_surface(layout_surface);
+        if (!shell->bkgnd_view) {
+            w_surface = lyt->surface_get_weston_surface(layout_surface);
+
+            alpha = ((shell->bkgnd_color >> 24) & 0xFF) / 255.0F;
+            red = ((shell->bkgnd_color >> 16) & 0xFF) / 255.0F;
+            green = ((shell->bkgnd_color >> 8) & 0xFF) / 255.0F;
+            blue = (shell->bkgnd_color & 0xFF) / 255.0F;
+
+            weston_surface_set_color(w_surface, red, green, blue, alpha);
+
+            wl_list_init(&shell->bkgnd_transform.link);
+            shell->bkgnd_view = weston_view_create(w_surface);
+            weston_layer_entry_insert(&shell->bkgnd_layer.view_list,
+                                      &shell->bkgnd_view->layer_link);
+        }
+
+        set_bkgnd_surface_prop(shell);
+    }
+    else {
+        wl_list_for_each(not, &ivisurf->notification_list, layout_link) {
+            ctrl = wl_resource_get_user_data(not->resource);
+            send_surface_event(ctrl, ivisurf->layout_surface, surface_id, ivisurf->prop,
+                               IVI_NOTIFICATION_CONFIGURE);
+        }
```

Comment:
Background surface could not be in notification list. Hmi controller does not know its existence.